types/coin.go

```diff
@@ -576,8 +588,8 @@ func (coins Coins) Sort() Coins {
 // Parsing
 
 var (
-	// Denominations can be 3 ~ 64 characters long.
-	reDnmString = `[a-z][a-z0-9/]{2,63}`
+	// Denominations can be 3 ~ 128 characters long.
+	reDnmString = `\p{L}[\p{L}\p{N}/_-]{2,127}`
```

Comment:
I don't understand this regex, can you document what it does?

Reply:
In addition this change goes from must start with a lower case letter followed by lowercase letters and digits... to what appears to be any valid letter followed by any valid letter or number.  Is the intent really to remove the lower case restriction then?

Reply:
updated